pkg/appender/store.go

```diff
@@ -400,7 +410,7 @@ func (cs *chunkStore) writeChunks(mc *MetricsCache, metric *MetricState) (hasPen
 			cs.pending = cs.pending[pendingSampleIndex:]
 		}
 
-		if pendingSamplesCount == 0 || expr == "" {
+		if pendingSamplesCount == 0 || expr == "" || err != nil {
```

Comment:
why do we need to check the error again here?

Reply:
yes, cause if we existed the loop with an error i want to push the metric back to queue